Missing <algorithm> in assemble_utils.cpp

Hi,

When running `ninja -j3 install -C build-dir`, the following error was raised:

```
dolfinx_mpc/cpp/assemble_utils.cpp:22:24: error: ‘find’ is not a member of ‘std’
```
I fixed that by adding `#include <algorithm>` in that file. Am I the only one who needed to add this `include`? Maybe it would be worth adding it for everyone.

Best,

Lucas

--- a/cpp/assemble_utils.cpp
+++ b/cpp/assemble_utils.cpp
@@ -5,6 +5,7 @@
 // SPDX-License-Identifier:    MIT
 
 #include "assemble_utils.h"
+#include <algorithm>
 #include <assert.h>
 std::vector<std::int32_t> dolfinx_mpc::compute_local_slave_index(
     std::span<const std::int32_t> slaves, const std::uint32_t num_dofs,

